quodlibet/quodlibet/ext/playorder/shufflebygrouping.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Phoenix Dailey
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import random
+
+from gi.repository import Gtk, GLib
+
+from quodlibet import _
+from quodlibet import app
+from quodlibet.order import OrderInOrder
+from quodlibet.order import OrderRemembered
+from quodlibet.order.reorder import Reorder
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.playorder import ShufflePlugin
+from quodlibet.qltk import Icons
+
+
+pconfig = PluginConfig("shufflebygrouping")
+pconfig.defaults.set("grouping", "~grouping~album~albumartist")
+pconfig.defaults.set("grouping_test", "grouping")
+pconfig.defaults.set("delay", 0)
+
+
+class ShuffleByGrouping(ShufflePlugin, OrderInOrder, OrderRemembered):
```

Comment:
Is `OrderInOrder` needed here (for a shuffle plugin)?

Reply:
I was going to say yes because this plugin actually disables shuffling if the next song belongs to the same group as the current song, but removing it and replacing `next_song = super().next(playlist, current)` with `next_song = OrderInOrder.next(self, playlist, current)` seems to work (perhaps because the body of OrderInOrder.next does not ever use `self`, so is essentially static).